### PR TITLE
Treat 401 as 404 with get_manifest

### PR DESCRIPTION
**Pull request by rcerven**
* OSBS-8217

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [ ] Commit messages are descriptive enough
- [ ] "Signed-off-by:" line is present in each commit
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by twaugh** [comment]
Might be better to do this in auth.py, here: https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/auth.py#L92

That way we know we're in the exact situation that:
- we originally got a 401 response
- then we obtained a bearer token
- but we still get a 401 response even with that

**Comment by rcerven** [comment]
> Might be better to do this in auth.py, here: https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/auth.py#L92
> 
> That way we know we're in the exact situation that:
> 
>     * we originally got a 401 response
> 
>     * then we obtained a bearer token
> 
>     * but we still get a 401 response even with that

changed

File: atomic_reactor/auth.py [hunk 1]
```diff
@@ -92,6 +92,10 @@ def handle_401(self, response, repo, **kwargs):
         retry_response = response.connection.send(retry_request, **kwargs)
         retry_response.history.append(response)
         retry_response.request = retry_request
+        # when quay returns 401 even after we have token,
+        # repository doesn't exist at all, so we will treat it as 404
+        if retry_response.status_code == requests.codes.unauthorized:
+            retry_response.status_code = requests.codes.not_found
```
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/auth.py]: just added test :)

[the remark this answers is not in the document]
Comment (twaugh) [reply atomic_reactor/auth.py]: Thanks.

**Comment by MartinBasti** [comment]
What happen when bearer token expires in our case? (I don't know validity period)

**Comment by rcerven** [comment]
@MartinBasti  if that would already happen we would see it,
because currently it handles 401 ONLY when it doesn't have token

**Comment by twaugh** [comment]
Maybe we should always add the `handle_401` hook, even if we already have a token? Would that handle expiry? Anyway, probably work for another PR. This PR doesn't break expiry, I think it's just not something that's handled.

**Comment by rcerven** [comment]
rn updated: - Treat 401 as 404 with registry request using BearerAuth

**Pull request merged**